karaf/itest/src/test/java/org/apache/brooklyn/rest/BrooklynRestApiLauncherTest.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.brooklyn.rest;
+
+import java.io.File;
+import java.util.concurrent.Callable;
+import org.apache.brooklyn.AssemblyTest;
+import org.apache.brooklyn.entity.brooklynnode.BrooklynNode;
+
+import org.apache.brooklyn.test.Asserts;
+import org.apache.brooklyn.util.http.HttpAsserts;
+import org.apache.brooklyn.util.http.HttpTool;
+import org.apache.http.HttpStatus;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.ops4j.pax.exam.Configuration;
+import static org.ops4j.pax.exam.CoreOptions.junitBundles;
+import org.ops4j.pax.exam.Option;
+import org.ops4j.pax.exam.junit.PaxExam;
+import static org.ops4j.pax.exam.karaf.options.KarafDistributionOption.configureConsole;
+import static org.ops4j.pax.exam.karaf.options.KarafDistributionOption.editConfigurationFilePut;
+import static org.ops4j.pax.exam.karaf.options.KarafDistributionOption.features;
+import static org.ops4j.pax.exam.karaf.options.KarafDistributionOption.karafDistributionConfiguration;
+import static org.ops4j.pax.exam.karaf.options.KarafDistributionOption.logLevel;
+import org.ops4j.pax.exam.karaf.options.LogLevelOption;
+import org.ops4j.pax.exam.spi.reactors.ExamReactorStrategy;
+import org.ops4j.pax.exam.spi.reactors.PerClass;
+
+@RunWith(PaxExam.class)
+@ExamReactorStrategy(PerClass.class)
+@Ignore // TODO: re-enable after brooklyn is properly initialized within the OSGI environment
```

Comment:
I've re-enabled the test after fixing the features for winrm4j, and here is the problem:

```
Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 10.586 sec <<< FAILURE! - in org.apache.brooklyn.rest.BrooklynRestApiLauncherTest
testStart(org.apache.brooklyn.rest.BrooklynRestApiLauncherTest)  Time elapsed: 1 sec  <<< FAILURE!
java.lang.AssertionError: URL http://localhost:9998/v1/catalog/entities does not contain text: BrooklynNode
        at org.apache.brooklyn.test.Asserts.fail(Asserts.java:721)
        at org.apache.brooklyn.util.http.HttpAsserts.assertContentContainsText(HttpAsserts.java:191)
        at org.apache.brooklyn.rest.BrooklynRestApiLauncherTest.testStart(BrooklynRestApiLauncherTest.java:89)
```

This is due to the catalog being empty, which is expected at this stage.